libs/@wingcloud/framework/src/internal.ts

```diff
@@ -6,32 +6,80 @@ export interface CompileOptions {
   entrypoint: string;
 }
 
+const MANAGED_TSCONFIG_OPTIONS = [
+  "target",
+  "module",
+  "moduleResolution",
+  "paths",
+  "outDir",
+];
+
 export async function compile(options: CompileOptions) {
+  const entrypointDir = dirname(options.entrypoint);
   const ts = (await import("typescript")).default;
   const outDir = join(options.workDir, "ts");
+  const wingsdkDir = dirname(require.resolve("@winglang/sdk"));
 
-  const program = ts.createProgram([options.entrypoint], {
+  const tsconfigPath = ts.findConfigFile(entrypointDir, ts.sys.fileExists);
+  const compilerOptions: ts.CompilerOptions = {
     target: ts.ScriptTarget.ES2022,
     module: ts.ModuleKind.CommonJS,
     moduleResolution: ts.ModuleResolutionKind.Node10,
+    paths: {
+      "@winglang/sdk": [wingsdkDir],
+      "@winglang/sdk/*": [wingsdkDir + "/*"],
+    },
+    outDir,
     alwaysStrict: true,
     allowSyntheticDefaultImports: true,
     esModuleInterop: true,
     strict: true,
     sourceMap: true,
-    outDir,
     declaration: false,
     noEmitOnError: true,
     listEmittedFiles: true,
     allowJs: true,
     skipLibCheck: true,
     resolveJsonModule: true,
     isolatedModules: true,
-    baseUrl: dirname(options.entrypoint),
-    paths: {
-      "@winglang/sdk/*": [dirname(require.resolve("@winglang/sdk")) + "/*"],
-    },
-  });
+  };
+
+  if (tsconfigPath) {
+    const { config, error } = ts.readConfigFile(tsconfigPath, ts.sys.readFile);
+    if (error) {
+      throw new Error(
+        `Failed to read tsconfig at ${tsconfigPath}: ${error.messageText}`
+      );
+    }
+
+    const { options, errors } = ts.parseJsonConfigFileContent(
+      config,
+      ts.sys,
+      dirname(tsconfigPath),
+      {}
+    );
+
+    if (errors.length > 0) {
+      throw new Error(
+        `Failed to parse tsconfig at ${tsconfigPath}: ${errors
+          .map((e) => e.messageText)
+          .join("\n")}`
+      );
+    }
+
+    for (const managedOption of MANAGED_TSCONFIG_OPTIONS) {
+      if (config?.compilerOptions?.[managedOption] !== undefined) {
+        console.warn(
+          `'${managedOption}' in tsconfig is managed by wing, value is ${compilerOptions[managedOption]}`
```

Comment:
Maybe add the expected value?

Reply:
Changed the wording to show what the forced options are. The goal is that users shouldn't/can't provide them at all.